src/integrations.py

```diff
@@ -58,6 +62,15 @@ def pop(self, key: str) -> dict:
         data = peers.data[self._app].pop(key, None)
         return json.loads(data) if data else {}
 
+    def prepare(self) -> None:
+        if not self[COOKIES_KEY]:
```

Comment:
Just a **NONBLOCKING** thought, not sure whether it's better if we do something like this:

```python
    def prepare(self) -> None:
        if not self._model.unit.is_leader():
            return
        
        if not self[COOKIES_KEY]:
            self[COOKIES_KEY] = secrets.token_hex(16)
```